website/static/js/recentActivityWidget.js

```diff
@@ -0,0 +1,325 @@
+        var nodeEvents = ((ctrl.eventNumbers.nodes/ctrl.totalEvents)*100 | 0) + (ctrl.eventNumbers.nodes ? 5 : 0);
+        var otherEvents = 100 - (fileEvents + commentEvents + wikiEvents + nodeEvents);
+        var filterLabels = function(){
+            if (!ctrl.eventFilter){
+                if (otherEvents === 100) {
+                    return m('p', 'No filters available');
+                }
+                return m('p', [
+                    'Filter on: ',
+                    fileEvents ? m('a', {onclick: function(){ctrl.callLogs('file')}}, 'Files' + (nodeEvents || commentEvents || wikiEvents ? ', ': '')): '', // jshint ignore:line
+                    nodeEvents ? m('a', {onclick: function(){ctrl.callLogs('project')}}, 'Projects' + (commentEvents || wikiEvents ? ', ': '')): '', // jshint ignore:line
+                    commentEvents ? m('a', {onclick: function(){ctrl.callLogs('comment')}}, 'Comments' + (wikiEvents ? ', ': '')): '',// jshint ignore:line
+                    wikiEvents ? m('a', {onclick: function(){ctrl.callLogs('wiki')}}, 'Wiki'): '' // jshint ignore:line
+                ]);
+            } else {
+                return m('p', [
+                    m('span','Filtering on '),
+                    m('b', (ctrl.eventFilter === 'file' ? 'Files' : ctrl.eventFilter === 'project' ? 'Projects' : ctrl.eventFilter === 'comment' ? 'Comments' : 'Wiki') + ' '),
+                    m('span.badge.pointer.m-l-xs', {
+                        onclick: function(){ ctrl.callLogs(ctrl.eventFilter); },
+                    }, [ m('i.fa.fa-close'), ' Clear'])
+                ]);
+            }
+        };
+        return m('.fb-activity-list.col-md-10.col-md-offset-1.m-t-xl', [
+                m('.time-slider-parent',
+                    m('#rASlider',  {config: ctrl.addSlider})
+                ),
+                m('canvas#rACanvas', {
+                    style: {verticalAlign: 'middle'},
+                    width: $('#rASlider').width(),
+                    height: ctrl.canvasHeight
+                }),
+                m('.row', [
+                    m('.col-xs-10.col-xs-offset-1',
+                        m('#rAProgressBar.progress.category-bar',
+                            ctrl.loading ? m('.progress-bar.progress-bar-success.active.progress-bar-striped', {style: {width: '100%'}}, m('b', {style:{color: 'white'}}, 'Loading')) : ([
+                                m('a.progress-bar' + (ctrl.eventFilter === 'file' ||  ctrl.eventFilter === false ?  '.selected' : ''), {style: {width: fileEvents+'%'},
+                                    onclick: function(){
+                                        ctrl.callLogs('file');
+                                    }}, m('i.fa.fa-file.progress-bar-button')
+                                ),
+                                m('a.progress-bar.progress-bar-warning' + (ctrl.eventFilter === 'project' ||  ctrl.eventFilter === false ?  '.selected' : ''), {style: {width: nodeEvents+'%'},
```

Comment:
Not super necessary but it looks like this can be unified into a template generating function 
```js
var progressBarTemplate = function (type){
    return m('a.progress-bar' + (ctrl.eventFilter === type ||  ctrl.eventFilter === false ?  '.selected' : ''), {style: {width: eventCount[type]+'%'},
     onclick: function(){
        ctrl.callLogs(type);
    }}, m('i.fa.fa-' + type + '.progress-bar-button')
    ),
}
```
It might need a bit more work than that but check it out.